apps/src/lib/client/tendermint_rpc_types.rs

```diff
@@ -34,83 +35,69 @@ pub struct TxResponse {
-                .unwrap();
-        let initialized_accounts = selector(&format!(
-            "$.events.['{}.initialized_accounts'][{}]",
-            evt_key, index
-        ));
-        let initialized_accounts = match initialized_accounts {
-            Ok(values) if !values.is_empty() => {
-                // In a response, the initialized accounts are encoded as e.g.:
-                // ```
-                // "applied.initialized_accounts": Array([
-                //   String(
-                //     "[\"atest1...\"]",
-                //   ),
-                // ]),
-                // ...
-                // So we need to decode the inner string first ...
-                let raw: String =
-                    serde_json::from_value(values[0].clone()).unwrap();
-                // ... and then decode the vec from the array inside the string
-                serde_json::from_str(&raw).unwrap()
-            }
-            _ => vec![],
-        };
-        TxResponse {
-            info: serde_json::from_value(info[0].clone()).unwrap(),
-            log: serde_json::from_value(log[0].clone()).unwrap(),
-            height: serde_json::from_value(height[0].clone()).unwrap(),
-            hash: serde_json::from_value(hash).unwrap(),
-            code: serde_json::from_value(code[0].clone()).unwrap(),
-            gas_used: serde_json::from_value(gas_used[0].clone()).unwrap(),
-            initialized_accounts,
+impl TryFrom<Event> for TxResponse {
+    type Error = String;
+
+    fn try_from(event: Event) -> Result<Self, Self::Error> {
+        fn missing_field_err(field: &str) -> String {
+            format!("Field \"{field}\" not present in event")
         }
+
+        let hash = event
+            .get("hash")
+            .ok_or_else(|| missing_field_err("hash"))?
+            .clone();
+        let info = event
+            .get("info")
+            .ok_or_else(|| missing_field_err("info"))?
+            .clone();
+        let log = event
+            .get("log")
+            .ok_or_else(|| missing_field_err("log"))?
+            .clone();
+        let height = event
+            .get("height")
+            .ok_or_else(|| missing_field_err("height"))?
+            .clone();
+        let code = event
+            .get("code")
+            .ok_or_else(|| missing_field_err("code"))?
+            .clone();
+        let gas_used = event
+            .get("gas_used")
+            .ok_or_else(|| missing_field_err("gas_used"))?
+            .clone();
+        let initialized_accounts = event
+            .get("initialized_accounts")
+            .map(String::as_str)
+            // TODO: fix finalize block, to return initialized accounts,
+            // even when we reject a tx?
+            .or(Some("[]"))
+            // NOTE: at this point we only have `Some(vec)`, not `None`
+            .ok_or_else(|| unreachable!())
+            .and_then(|initialized_accounts| {
```

Comment:
```suggestion
            .map_or(Ok(vec![]), |initialized_accounts| {
```

Reply:
really I'd much rather do this https://github.com/anoma/namada/issues/20, but this is better than before anyway